test/zone-spec/fake-async-test.spec.ts

```diff
@@ -0,0 +1,307 @@
+import '../../lib/zone-spec/fake-async-test';
+import {ifEnvSupports} from '../test-util';
+
+describe('FakeAsyncTestZoneSpec', () => {
+  let FakeAsyncTestZoneSpec = Zone['FakeAsyncTestZoneSpec'];
+  let testZoneSpec;
+  let fakeAsyncTestZone;
+
+  beforeEach(() => {
+    testZoneSpec = new FakeAsyncTestZoneSpec('name');
+    fakeAsyncTestZone = Zone.current.fork(testZoneSpec);
+  });
+  
+  it('sets the FakeAsyncTestZoneSpec property', () => {
+    fakeAsyncTestZone.run(() => {
+      expect(Zone.current.get('FakeAsyncTestZoneSpec')).toEqual(testZoneSpec);
+    });
+  });
+
+  describe('synchronous code', () => {
+    it('should run', () => {
+      let ran = false;
+      fakeAsyncTestZone.run(() => { ran = true; });
+
+      expect(ran).toEqual(true);
+    });
+
+    it('should throw the error in the code', () => {
+      expect(() => {
+        fakeAsyncTestZone.run(() => { throw new Error('sync'); });
+      }).toThrowError('sync');
+    });
+  });
+
+  describe('asynchronous code', () => {
+    it('should run', () => {
+      fakeAsyncTestZone.run(() => {
+        let thenRan = false;
+        Promise.resolve(null).then((_) => { thenRan = true; });
+
+        expect(thenRan).toEqual(false);
+
+        testZoneSpec.flushMicrotasks();
+        expect(thenRan).toEqual(true);
+      });
+    });
+
+    it('should rethrow the exception on flushMicroTasks for error thrown in Promise callback',
+      () => {
+        fakeAsyncTestZone.run(() => {
+          Promise.resolve(null).then((_) => { throw new Error('async'); });
+          expect(() => { testZoneSpec.flushMicrotasks(); })
+            .toThrowError('Uncaught (in promise): Error: async');
+        });
+      });
+
+    it('should run chained thens', () => {
+      fakeAsyncTestZone.run(() => {
+        let log = [];
+
+        Promise.resolve(null).then((_) => log.push(1)).then((_) => log.push(2));
+
+        expect(log).toEqual([]);
+
+        testZoneSpec.flushMicrotasks();
+        expect(log).toEqual([1, 2]);
+      });
+    });
+
+    it('should run Promise created in Promise', () => {
+      fakeAsyncTestZone.run(() => {
+        let log = [];
+
+        Promise.resolve(null).then((_) => {
+          log.push(1);
+          Promise.resolve(null).then((_) => log.push(2));
+        });
+
+        expect(log).toEqual([]);
+
+        testZoneSpec.flushMicrotasks();
+        expect(log).toEqual([1, 2]);
+      });
+    });
+  });
+
+  describe('timers', () => {
+    it('should run queued zero duration timer on zero tick', () => {
+      fakeAsyncTestZone.run(() => {
+        let ran = false;
+        setTimeout(() => { ran = true }, 0);
+
+        expect(ran).toEqual(false);
+
+        testZoneSpec.tick();
```

Comment:
Question: is `tick()` just an alias for `tick(0)` - and what was the previous behavior? Or should it try to clear out any non-periodic timers?